### PR TITLE
Fetch sorted workflows based on TagResources

### PR DESCRIPTION
**Pull request by mkanoor**
Catalog would be sending a collection of TagResources and this
code will return a list of workflows sorted by the sequence number

### CODE REVIEW

**Comment by mkanoor** [comment]
The travis failure is happening because the strong parameter check is dropping the tag_resource parameters since it's an array of hashes.
https://github.com/ManageIQ/approval-api/blob/c2725503a8a2f8fa408c5253b82b1e2d6b5158f9/app/controllers/api/v1x0/requests_controller.rb#L35

**Comment by mkanoor** [comment]
@bzwei please review

File: app/services/workflow_find_service.rb [hunk 1]
```diff
@@ -8,8 +8,21 @@ def find(tag_attrs)
...
+      tag_names = tr['tags'].map { |tag| "/#{tag['namespace']}/#{tag['name']}=#{tag['value']}" }
+      params = {:app_name => tr['app_name'], :object_type => tr['object_type'], :tag_name => tag_names}
+      query =
+        if i.zero?
+          query.where(params)
+        else
+          query.or(TagLink.where(params))
+        end
+    end
+    Workflow.where(:id => query).order(:sequence)
```
Comment (bzwei) [inline app/services/workflow_find_service.rb]: No need to order by sequence. The default scope already has it.

File: spec/services/workflow_find_service_spec.rb [hunk 2]
```diff
@@ -30,4 +30,23 @@
       expect(another_workflows).to eq([])
     end
   end
+
+  describe 'find_by_tag_resources' do
```
Comment (bzwei) [inline spec/services/workflow_find_service_spec.rb]: describe '#find_by_tag_resources'

File: spec/services/workflow_find_service_spec.rb [hunk 3]
```diff
@@ -30,4 +30,23 @@
...
+    include_context "tag_resource_objects"
+
+    it 'finds sorted workflow based on tags' do
+      workflows = subject.find_by_tag_resources([tag_resource1])
+      expect(workflows).to eq([workflow2, workflow1])
+    end
+
+    it 'if tags is empty returns empty array' do
```
Comment (bzwei) [inline spec/services/workflow_find_service_spec.rb]: better reads `it 'returns an empty array if tags are empty'`

File: spec/services/workflow_find_service_spec.rb [hunk 4]
```diff
@@ -30,4 +30,23 @@
...
+      workflows = subject.find_by_tag_resources([tagless_resource])
+      expect(workflows).to be_empty
+    end
+
+    it 'if input is empty array return empty array' do
```
Comment (bzwei) [inline spec/services/workflow_find_service_spec.rb]: better reads `it 'returns an empty array if input is empty'`

File: spec/services/workflow_find_service_spec.rb [hunk 5]
```diff
@@ -30,4 +30,23 @@
...
+      workflows = subject.find_by_tag_resources([])
+      expect(workflows).to be_empty
+    end
+  end
```
Comment (bzwei) [inline spec/services/workflow_find_service_spec.rb]: can we have a test `it 'returns an empty array if tags find no links`?

**Pull request merged**